st2api/st2api/controllers/v1/datastore.py

```diff
@@ -34,15 +34,22 @@ class KeyValuePairController(RestController):
     """
 
     @jsexpose(str)
-    def get_one(self, id):
+    def get_one(self, name):
```

Comment:
At some point there will be proper RBAC in our system. The key-value store we offer will also have RBAC on top therefore the user RBAC privileges will determine the space of keys that a user can see. By using name we end up in a weird situation where user John and user Jerry both try to save  a key aws_creds and one will get a failure. We don't have clear requirements yet on this thing but atleast using a system generated ID as the unique identifier keys us somewhat isolated from the problem until we need to add such a feature.